node/src/actors/chain_manager/mod.rs

```diff
@@ -341,6 +351,37 @@ fn update_transaction_pool(transactions_pool: &mut TransactionsPool, transaction
     }
 }
 
+/// Update the list of unspent outputs that can be spent by this node.
+fn update_own_utxos(
+    own_utxos: &mut HashSet<OutputPointer>,
+    own_pkh: PublicKeyHash,
+    transactions: &[Transaction],
+) {
```

Comment:
*Suggestion*: I would prefer a function that filter all the transactions with output `own_pkh` and returns an iterator to those values, later we could use it to update the utxos. The idea is to keep the *side-effecting* as contained as possible

Reply:
It looks to me like this should be affected by a diffing strategy too. Instead of mutating `own_utxos` here, we should accumulate `insert(output_pointer)` and `delete(output_pointer)` commands into a vector just as we are validating the transactions, with views to later apply those changes atomically when we consolidate the block.